### ISSUE TITLE
vmselect with clusternativeListenAddr option crashes when any one underlying storage node crashes 

### ISSUE THREAD

**Issue by sharninder-se** [VictoriaMetrics/VictoriaMetrics#2961]
**Describe the bug**
vmselect running with the clusternativeListenAddr option and denyPartionalResponse=false, replicationFactor=1 crashes if a query is made to vmselect and any one vmstorage nodes is unavailable.
 
**To Reproduce**
Just make a query to vmselect with the described configuration.

**Expected behavior**
vmselect should not crash.

**Logs**
This is the stack trace I see:

```
{"ts":"2022-08-08T10:53:30.427Z","level":"info","caller":"VictoriaMetrics/lib/vmselectapi/server.go:170","msg":"closing vmselect conn from 10.102.2.163:55262"}
panic: send on closed channel

goroutine 127 [running]:
github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/clusternative.newBlockIterator.func1.1(0xc000128100)
	github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/clusternative/vmselect.go:120 +0x59
github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/netstorage.(*storageNode).processSearchQueryOnConn(0xc0001b4000, 0xc00033c780, {0xc000140e00?, 0x1?, 0x1?}, 0xc000056db0)
	github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/netstorage/netstorage.go:2191 +0x28f
github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/netstorage.(*storageNode).processSearchQuery.func1(0x0?)
	github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/netstorage/netstorage.go:1651 +0x33
github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/netstorage.(*storageNode).execOnConn(0xc0001b4000, 0x0, {0x90e183, 0x9}, 0xc000054e60, {0x62f0eb68, 0xe56101074, {0x0, 0x0}})
	github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/netstorage/netstorage.go:1725 +0x69d
github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/netstorage.(*storageNode).execOnConnWithPossibleRetry(0x0?, 0x0?, {0x90e183, 0x9}, 0x7f519c17d878?, {0x10?, 0xc000210400?, {0x0?, 0x0?}})
	github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/netstorage/netstorage.go:1661 +0xff
github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/netstorage.(*storageNode).processSearchQuery(0x0?, 0x0?, {0xc000140e00?, 0x0?, 0x0?}, 0x0?, {0x62f0eb68, 0xe56101074, {0x0, 0x0}})
	github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/netstorage/netstorage.go:1656 +0x8d
github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/netstorage.ProcessBlocks.func1(0x0?, 0x0?, 0xc0001b4000)
	github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/netstorage/netstorage.go:1350 +0x105
github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/netstorage.startStorageNodesRequest.func1(0x0?, 0x0?)
	github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/netstorage/netstorage.go:1379 +0x39
created by github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/netstorage.startStorageNodesRequest
	github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/netstorage/netstorage.go:1378 +0x76
```

**Comment by valyala** [comment]
@sharninder-se , thanks for filing the issue. It should be fixed in the commit 1996e36cf0638b30573ce79b4bd87f7ca36a10ea , which will be included in the next release. In the mean time you can build `vmselect` from this commit according to [these docs](https://docs.victoriametrics.com/Cluster-VictoriaMetrics.html#building-from-sources) and verify whether the issue is fixed there.

**Comment by valyala** [comment]
The bug should be fixed in [VictoriaMetrics v1.79.2](https://github.com/VictoriaMetrics/VictoriaMetrics/releases). Closing the issue as fixed.

@sharninder-se , feel free re-opening the issue if it still exists in VictoriaMetrics v1.79.2 .
